packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -227,13 +263,29 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
     }
   }
 
+  WidgetStateProperty<Color?> get _widgetFillColor {
+    return WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+      if (states.contains(WidgetState.disabled)) {
+        return widget.inactiveColor;
+      }
+      if (states.contains(WidgetState.selected)) {
+        return widget.activeColor;
+      }
+      return null;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
-    final Color effectiveActiveColor = widget.activeColor
-      ?? CupertinoColors.activeBlue;
-    final Color? inactiveColor = widget.inactiveColor;
-    final Color effectiveInactiveColor = inactiveColor
-      ?? CupertinoColors.inactiveGray;
+     // Colors need to be resolved in selected and non selected states separately
+    final Set<WidgetState> activeStates = states..add(WidgetState.selected);
+    final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
+
+    final Color? activeColor = widget.fillColor?.resolve(activeStates) ?? _widgetFillColor.resolve(activeStates);
```

Comment:
It might be a little confusing to have an `activeColor` here when the expectation is `activeColor` should be `widget.activeColor`. I'd recommend consolidating this `activeColor` into `effectiveActiveColor`, so it's clear what all variables are referring to.

Reply:
We are already using `effectiveActiveColor`, will it be bad if we call it `resolvedActiveColor` ?

Reply:
Reason we are having `activeColor` and `effectiveActiveColor` is `fillColor` and `_widgetFillColor` might resolve to null.
So, either we can rename `activeColor` to `resolvedActiveColor`
or
Just keep `effectiveActiveColor` that resolves everything